packages/desktop/components/popups/send/SendConfirmationPopup2.svelte

```diff
@@ -0,0 +1,235 @@
+        if (Number(surplus) > 0) {
+            if (minimumStorageDeposit >= Number(surplus)) {
+                visibleSurplus = surplus = undefined
+            } else {
+                visibleSurplus = Number(surplus) - minimumStorageDeposit
+                // Note: we have to hide it because currently, in the sdk,
+                // the storage deposit return strategy is only looked at
+                // if the provided amount is < the minimum required storage deposit
+                hideGiftToggle = true
+            }
+        }
+    }
+
+    function toggleGiftStorageDeposit(): void {
+        giftStorageDeposit = !giftStorageDeposit
+    }
+
+    async function sendOutputAndClosePopup(): Promise<void> {
+        await sendOutput(preparedOutput)
+        closePopup()
+    }
+
+    async function onConfirmClick(): Promise<void> {
+        try {
+            validateSendConfirmation(preparedOutput as CommonOutput)
+
+            if ($isActiveLedgerProfile) {
+                ledgerPreparedOutput.set(preparedOutput)
+            }
+
+            updatePopupProps({ isSendAndClosePopup: true, preparedOutput, calculatedStorageDeposit: storageDeposit })
+            await checkActiveProfileAuth(sendOutputAndClosePopup, { stronghold: true, ledger: false })
+        } catch (err) {
+            handleError(err)
+        }
+    }
+    function onBackClick(): void {
+        closePopup()
+        openPopup({
+            id: PopupId.SendForm,
+            overflow: true,
+        })
+    }
+
+    function onCancelClick(): void {
+        closePopup()
+    }
+</script>
+
+<send-confirmation-popup class="w-full h-full space-y-6 flex flex-auto flex-col shrink-0">
+    <Text type={TextType.h3} fontWeight={FontWeight.semibold} classes="text-left"
+        >{localize('popups.transaction.title')}</Text
+    >
+    <div class="w-full flex-col space-y-2">
+        {#if transactionDetails.type === NewTransactionType.TokenTransfer}
+            <BasicActivityDetails {activity} />
+        {:else if transactionDetails.type === NewTransactionType.NftTransfer}
+            <NftActivityDetails {activity} />
+        {/if}
+        <div class="pt-4">
+            <ActivityInformation {activity} bind:activeTab />
+        </div>
+        {#if activeTab === Tab.Transaction}
+            {#if !hideGiftToggle}
+                <KeyValueBox keyText={localize('general.giftStorageDeposit')}>
+                    <Toggle
+                        slot="value"
+                        color={ToggleColor.Green}
+                        disabled={disableToggleGift}
+                        active={giftStorageDeposit}
+                        onClick={toggleGiftStorageDeposit}
+                    />
+                </KeyValueBox>
+            {/if}
+            {#if initialExpirationDate !== undefined}
+                <KeyValueBox keyText={localize('general.expirationTime')}>
+                    <ExpirationTimePicker
```

Comment:
When sending native tokens toggling git storage to `true` sets Expiration time to `None` if user than clicks to pick new expiration time in the opened time picker last value  (in this example `In 1 week`) is selected instead of `No expiration time`